olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/StreamLayerClient.h

```diff
@@ -86,8 +94,35 @@ class DATASERVICE_READ_API StreamLayerClient final {
 
   ~StreamLayerClient();
 
+  /**
+   * @brief Cancels all active and pending requests.
+   * @return True on success.
+   */
+  bool CancelPendingRequests();
+
+  /**
+   * @brief Enables message consumption for the specific stream layer.
+   * @param request contains the complete set of request parameters.
+   * @param callback will be invoked once the subscription request is completed,
+   * or an error is encountered.
+   * @return A token that can be used to cancel this request
+   */
+  client::CancellationToken Subscribe(
+      SubscribeRequest request,
+      SubscribeRequest::SubscribeResponseCallback callback);
+
+  /**
+   * @brief Enables message consumption for the specific stream layer.
+   * @param request contains the complete set of request parameters.
+   * @return CancellableFuture, which when complete will contain the
+   * SubscribeId or an error. Alternatively, the CancellableFuture can be
+   * used to cancel this request.
+   */
+  client::CancellableFuture<SubscribeRequest::SubscribeResponse> Subscribe(
+      SubscribeRequest request);
+
  private:
-  std::unique_ptr<StreamLayerClientImpl> impl_;
+  std::shared_ptr<StreamLayerClientImpl> impl_;
```

Comment:
Why shared_ptr? Being a single user class I wonder what is the logic behind it?

Reply:
Reverted.